Stratis.Bitcoin.Tests/BlockPulling/PullerDownloadAssignmentsTest.cs

```diff
@@ -107,5 +107,67 @@ public void AssignBlocksToPeersWithNodesWithDifferentChainsCorrectlyDistributesD
             }
             Assert.Equal(requiredBlockHeights.Count, tasksAssigned);
         }
+
+        /// <summary>
+        /// Very similar test to AssignBlocksToPeersWithNodesWithDifferentChainsCorrectlyDistributesDownloadTasks
+        /// except that this one uses randomized initialization and much larger sample.
+        /// </summary>
+        [Fact]
+        public void AssignBlocksToPeersLargeSampleCorrectlyDistributesDownloadTasks()
+        {
+            Random rnd = new Random();
+
+            int iterationCount = 1000;
+            for (int iteration = 0; iteration < iterationCount; iteration++)
+            {
+                // Choose scenario for this iteration.
+                int ourBlockCount = rnd.Next(1000);
+                int bestChainBlockCount = ourBlockCount + rnd.Next(1000);
+                int availablePeerCount = rnd.Next(100) + 1;
+
+                List<int> requiredBlockHeights = new List<int>();
+                for (int i = ourBlockCount + 1; i <= bestChainBlockCount; i++)
+                    requiredBlockHeights.Add(i);
+
+                requiredBlockHeights = requiredBlockHeights.OrderBy(a => rnd.Next()).ToList();
+
+                // Initialize node's peers.
+                int maxPeerChainLength = 0;
+                List<PullerDownloadAssignments.PeerInformation> availablePeersInformation = new List<PullerDownloadAssignments.PeerInformation>();
+                for (int peerIndex = 0; peerIndex < availablePeerCount; peerIndex++)
+                {
+                    PullerDownloadAssignments.PeerInformation peerInfo = new PullerDownloadAssignments.PeerInformation()
+                    {
+                        ChainHeight = rnd.Next(bestChainBlockCount) + 1,
+                        PeerId = peerIndex,
+                        QualityScore = rnd.Next(BlockPuller.MaxQualityScore) + 1
+                    };
+                    availablePeersInformation.Add(peerInfo);
+                    maxPeerChainLength = Math.Max(maxPeerChainLength, peerInfo.ChainHeight);
+                }
+
+                // Use the assignment strategy to assign tasks to peers.
+                Dictionary<PullerDownloadAssignments.PeerInformation, List<int>> assignments = PullerDownloadAssignments.AssignBlocksToPeers(requiredBlockHeights, availablePeersInformation);
+
+                // Check the assignment is valid per our requirements.
+                int tasksAssigned = 0;
+                Assert.Equal(availablePeersInformation.Count, assignments.Count);
+                foreach (KeyValuePair<PullerDownloadAssignments.PeerInformation, List<int>> kvp in assignments)
```

Comment:
I would be OK with that (although I prefer the full form), but using the var here would violate the current code style policy (actually that part of the current code style I like very much, because I don't consider var as being a good friend almost in any case), so please remember that for code style policy redesign if you want this. Maybe we should open a wide discussion somewhere (issue in github) where everyone could express their preference and after some time we could converge to a new code style policy.